core/domain/suggestion_services_test.py

```diff
@@ -1007,6 +1003,198 @@ def test_get_translation_suggestion_ids_with_exp_ids_with_empty_exp_list(
+        resubmitted, we count the time that the suggestion has been waiting for
+        review from when it was resubmitted, not from when it was first
+        submitted.
+        """
+        # Create a Hindi translation suggestion associated with exploration id
+        # target_id_1.
+        with self.swap(
+            exp_fetchers, 'get_exploration_by_id',
+            self.mock_get_exploration_by_id):
+            with self.swap(
+                exp_domain.Exploration, 'get_content_html',
+                self.MockExploration.get_content_html):
+                suggestion_1 = suggestion_services.create_suggestion(
+                    suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                    suggestion_models.TARGET_TYPE_EXPLORATION,
+                    self.target_id_1, 1, self.author_id_1,
+                    self.add_translation_change_dict, 'test description')
+        # Create a Hindi translation suggestion associated with exploration id
+        # target_id_2.
+        with self.swap(
+            exp_fetchers, 'get_exploration_by_id',
+            self.mock_get_exploration_by_id):
+            with self.swap(
+                exp_domain.Exploration, 'get_content_html',
+                self.MockExploration.get_content_html):
+                suggestion_2 = suggestion_services.create_suggestion(
+                    suggestion_models.SUGGESTION_TYPE_TRANSLATE_CONTENT,
+                    suggestion_models.TARGET_TYPE_EXPLORATION,
+                    self.target_id_2, 1, self.author_id_1,
+                    self.add_translation_change_dict, 'test description')
+
+        # Verify that both suggestions are returned and in the right order.
+        suggestions = (
+            suggestion_services
+            .get_translation_suggestions_waiting_longest_for_review_per_lang(
+                'hi'))
+        self.assertEqual(len(suggestions), 2)
+        self.assertLessEqual(
+            suggestions[0].suggestion_id, suggestion_1.suggestion_id)
+        self.assertLessEqual(
+            suggestions[1].suggestion_id, suggestion_2.suggestion_id)
+
+        # Reject the suggestion that was created first since it is the one that
+        # has been waiting the longest for review.
+        suggestion_services.reject_suggestion(
+            suggestion_1.suggestion_id, self.reviewer_id_1, 'Reject message')
+
+        # Verify that only the suggestion that was created second is returned.
+        suggestions = (
+            suggestion_services
+            .get_translation_suggestions_waiting_longest_for_review_per_lang(
+                'hi'))
+        self.assertEqual(len(suggestions), 1)
+        self.assertLessEqual(
+            suggestions[0].suggestion_id, suggestion_2.suggestion_id)
+
+        # Resubmit rejected suggestion.
+        with self.swap(
+            suggestion_registry.SuggestionTranslateContent,
+            'pre_update_validate',
+            self.mock_pre_update_validate_does_nothing):
```

Comment:
Why do we need to mock part of the validation flow? This seems slightly like code smell to me--is this case not actually valid in production if validation would fail? In general, tests should mimic valid production scenarios.

Reply:
Thanks--I think this is much clearer.